dbt/adapters/duckdb/environments/__init__.py

```diff
@@ -166,7 +166,10 @@ def initialize_db(
         # connection that they need to do
         if plugins:
             for plugin in plugins.values():
-                plugin.configure_connection(conn)
+                # Only connect to MotherDuck if this is not already
```

Comment:
Ah yes - so I had to add this if-statement because when I was running my test and connecting to a MotherDuck database (e.g. `"md:plugin_test"`, `MD_CONNECT` actually failed with the message `"already connected"`. This is because upon connecting to the database the MotherDuck extension is already loaded + database is attached automatically. I'll clarify the docstring a bit.

Reply:
But you are right that this is not the right place for that to live - I'll move it to the DuckDB plugin specifically

Reply:
gotcha, that totally makes sense 👍

Reply:
Looks like we don't need `MD_CONNECT` after all, just loading the extension and setting the token is sufficient!

Reply:
Hmm, so I think this fix is now failing in CI and I'm trying to understand why. It does seem to work for me both locally and on the `motherduckdb` fork (see [CI run](https://github.com/motherduckdb/dbt-duckdb/actions/runs/7746950669)) but it's been consistently failing on [this line](https://github.com/duckdb/dbt-duckdb/actions/runs/7746441003/job/21125011980#step:5:351) with the following exception:

```python
>           conn = duckdb.connect(creds.path, read_only=False, config=config)
E           duckdb.duckdb.InvalidInputException: Invalid Input Error: Initialization function "motherduck_init" from file 
"/home/runner/.duckdb/extensions/v0.9.2/linux_amd64_gcc4/motherduck.duckdb_extension" threw an exception: 
"Invalid Input Error: Initialization function "motherduck_init" from file 
"/home/runner/.duckdb/extensions/v0.9.2/linux_amd64_gcc4/motherduck.v1.15.10.duckdb_extension" threw an 
exception: "Unknown Type Error: Cannot connect to MotherDuck server: no token provided.""

/home/runner/work/dbt-duckdb/dbt-duckdb/dbt/adapters/duckdb/environments/__init__.py:142: InvalidInputException
```

I'm not sure why. Either way it seems that both for me locally and for the `motherduckdb` CI, the token is passed via environment variable and the auth to MotherDuck is done upon connection, while in this CI it needs an explicit call to `MD_CONNECT`.

Reply:
It works fine for me locally as well-- it may simply be that your PR cannot access the GH secret I use for my MD token.